src/main/java/random_csapatnev/BearVirus.java

```diff
@@ -39,27 +39,27 @@ public void affect(Character c) {
 		if (!hasProtection) {
 			c.currField.removeCharacter(c);
 
-			ArrayList<GraphicsCharacter> tempList = new ArrayList<>(MainFrame.Instance.model.graphicsCharacter.size());
-			for (GraphicsCharacter e : MainFrame.Instance.model.graphicsCharacter) {
+			ArrayList<GraphicsCharacter> tempList = new ArrayList<>(MainFrame.Instance.model.getGraphicsCharacter().size());
```

Comment:
ArrayList<GraphicsCharacter> tempList = ArrayList<>(MainFrame.Instance.getGraphicsCharacter()) modon eleg inicializalni, es akkor nincs szukseg a for ciklusra az elemek masolasahoz.

Reply:
[04f51b3](https://github.com/BME-MIT-IET/iet-hf-2023-grafadatbazis/pull/25/commits/04f51b398d995fc906f08a2a0682e87214df2a14) commitban javítva